sdk/src/trace/tracer_provider.cc

```diff
@@ -19,11 +19,29 @@ TracerProvider::TracerProvider(std::unique_ptr<SpanProcessor> processor,
                                                      std::move(id_generator)))
 {}
 
-opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> TracerProvider::GetTracer(
+nostd::shared_ptr<opentelemetry::trace::Tracer> TracerProvider::GetTracer(
     nostd::string_view library_name,
     nostd::string_view library_version) noexcept
 {
-  return opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer>(tracer_);
+  // if (library_name == "") {
+  //   // TODO: log invalid library_name.
+  // }
+
+  const std::lock_guard<std::mutex> guard(lock_);
+
+  for (auto &tracer : tracers_)
```

Comment:
Nit: Is it somehow possible to avoid this search loop by using a map instead of a vector say something like:
std::map<std::pair<name, version>, tracer> tracers_;

Reply:
I assumed there would not be too many tracers in one process (like less than a hundred) so linear scan is fine compared to the hash or sort overhead. But we could discuss this a little more.

Reply:
Yes, I too think so. If it's not a major performance benefit with std::map, and it adds more complexity to current design, we should be using vector.